mono/mini/aot-compiler.c

```diff
@@ -8291,7 +8291,19 @@ emit_llvm_file (MonoAotCompile *acfg)
 	if (acfg->aot_opts.llvm_only) {
 		/* Use the stock clang from xcode */
 		// FIXME: arch
-		command = g_strdup_printf ("clang++ -fexceptions -march=x86-64 -fpic -msse -msse2 -msse3 -msse4 -O2 -fno-optimize-sibling-calls -Wno-override-module -c -o \"%s\" \"%s.opt.bc\"", acfg->llvm_ofile, acfg->tmpbasename);
+		shared_flags = g_strdup_printf("-O2 -fexceptions -fpic -fno-optimize-sibling-calls -Wno-override-module -c -o \"%s\" \"%s.opt.bc\"", acfg->llvm_ofile, acfg->tmpbasename);
+    command = NULL;
+
+#if TARGET_ARM
+		command = g_strdup_printf ("clang++ -march=armv7 -mfpu=vfp3 %s", shared_flags);
```

Comment:
This won't work on Android for example. I know it isn't relevant now... but it might be at some point. We have the proper defines in place, like we do here:
https://github.com/mono/mono/blob/16e6e7df70a7b8fc3579e5f960d0610a313048ec/mono/mini/aot-compiler.c#L1076-L1084